app/models/message.rb

```diff
@@ -13,6 +13,11 @@ def self.human_attribute_name(attr, options = {})
     HUMANIZED_ATTRIBUTES[attr.to_sym] || super
   end
 
+  def self.find_active_by_token(token)
+    message = Message.find_by(token: token)
+    message unless message && message.expired?
+  end
```

Comment:
looks good! one thing on my mind is whether we should change this method so that it also deletes the message if it's expired.